core/src/main/java/com/linecorp/armeria/server/ServiceConfig.java

```diff
@@ -474,6 +474,16 @@ List<ShutdownSupport> shutdownSupports() {
         return shutdownSupports;
     }
 
+    /**
+     * Returns whether the access log should be written for this {@link #service()}.
+     * Access log is written only when the {@link TransientServiceOption#WITH_ACCESS_LOGGING} option is enabled
+     * and the {@link #accessLogWriter()} is not {@link AccessLogWriter#DISABLED}.
+     */
+    public boolean shouldWriteAccessLog() {
+        return !accessLogWriter.isDisabled() &&
+                transientServiceOptions.contains(TransientServiceOption.WITH_ACCESS_LOGGING);
+    }
```

Comment:
Should we create `AccessLogWriterUtil` and move this method to it? I don't think this ㅡethod needs to be a public API.

Reply:
sg, I moved the `maybeWriteAccessLog` entirely